### PR TITLE
Clean up path validation related issues in System.Drawing.Common

### PR DESCRIPTION
**Pull request by mellinoe**
There were several places in the .NET Framework codebase for System.Drawing which called `IntSecurity.DemandReadFileIO(string);`. On first glance, this looks to be a CAS-related call, which isn't interesting in .NET Core. What isn't obvious is that this also ends up calling `Path.GetFullPath` on the string passed in, and that call may throw argument validation exceptions. If the path is `null`, for example, the security demand will result in an ArgumentNullException. These calls were omitted from the .NET Core version when I trimmed out all of the security demands.

I've added these back, and fixed up several test cases which validated the behavior, and were previously disabled. Path.GetFullPath in .NET Core now correctly sets the `ParamName` property on its ArgumentException, so the assertions have been augmented to expect that.

There were several tests which also expected some methods to result in a `PathTooLongException` being thrown from `Path.GetFullPath`. However, .NET Core no longer eagerly throws that exception from `Path.GetFullPath`. I have removed those test cases.

@hughbe @KostaVlev

### CODE REVIEW

**Comment by mellinoe** [comment]
Fixes https://github.com/dotnet/corefx/issues/22723, https://github.com/dotnet/corefx/issues/22598, https://github.com/dotnet/corefx/issues/22367, https://github.com/dotnet/corefx/issues/22640.

**Comment by mellinoe** [comment]
@dotnet-bot Test NETFX x86 Release Build

File: src/System.Drawing.Common/tests/Imaging/ImageAttributesTests.cs [hunk 1]
```diff
@@ -1223,18 +1222,6 @@ public void SetOutputChannelColorProfile_InvalidPath_ThrowsOutOfMemoryException(
             }
         }
 
-        [ActiveIssue(22309)]
-        [ConditionalFact(Helpers.GdiplusIsAvailable)]
-        public void SetOutputChannelColorProfile_InvalidPath_ThrowsPathTooLongException()
```
Comment (KostaVlev) [inline src/System.Drawing.Common/tests/Imaging/ImageAttributesTests.cs]: If it makes sense we can keep this test if we change 
`string fileNameTooLong = new string('a', 261);` to `string fileNameTooLong = new string('a', short.MaxValue);`

File: src/System.Drawing.Common/tests/Imaging/MetafileTests.cs [hunk 2]
```diff
@@ -481,25 +478,9 @@ public void Ctor_InvalidPathI_ThrowsArgumentException(string fileName)
...
-            }
-        }
-
-        [ActiveIssue(22723)]
-        [ActiveIssue(20884, TestPlatforms.AnyUnix)]
-        [ConditionalFact(Helpers.GdiplusIsAvailable)]
-        public void Ctor_PathTooLong_ThrowsPathTooLongException()
-        {
-            string fileName = GetPath(new string('a', 261));
-            using (var bufferGraphics = Graphics.FromHwndInternal(IntPtr.Zero))
```
Comment (KostaVlev) [inline src/System.Drawing.Common/tests/Imaging/MetafileTests.cs]: Here also if it makes sense we can keep it if have string `fileName = GetPath(new string('a', shrot.MaxValue));`.

**Comment by mellinoe** [comment]
@KostaVlev I added back those two tests, using the larger string.

**Comment by mellinoe** [comment]
@dotnet-bot Test this please

(Lots of Jenkins communication issues)

**Comment by mellinoe** [comment]
@dotnet-bot Test Linux x64 Release Build

**Pull request merged**